tests/integration/cluster-users/sar-test.sh

```diff
@@ -48,31 +48,111 @@ function __deleteTestNamespace() {
 }
 
 function __createRoleBindingForNamespaceDeveloper() {
+	local result=1
 	set +e
 	kubectl create rolebinding 'namespace-developer' --clusterrole='kyma-developer' --user="${DEVELOPER_EMAIL}" -n "${CUSTOM_NAMESPACE}"
-	result = $?
+	result=$?
 	set -e
 
-	if [ result -eq 0 ]; then
+	if [[ ${result} -eq 0 ]]; then
 		echo "----> PASSED"
 		return 0
 	fi
 
 	echo "----> |FAIL|"
+	return 1
 }
 
 function __createNamespaceForNamespaceAdmin() {
+	local result=1
 	set +e
 	kubectl create namespace "${CUSTOM_NAMESPACE}"
-	result = $?
+	result=$?
 	set -e
-  
-	if [ result -eq 0 ]; then
+
+	if [[ ${result} -eq 0 ]]; then
 		echo "----> PASSED"
 		return 0
 	fi
-  
+
 	echo "----> |FAIL|"
+	return 1
+}
+
+function __testPermissions() {
+	local OPERATION="$1"
+	local RESOURCE="$2"
+	local TEST_NS="$3"
+	local EXPECTED="$4"
+	local TEST="not-set-yet"
+
+	if [[ "${TEST_NS}" != "--all-namespaces" ]]; then
+		TEST_NS="-n${TEST_NS}"
+	fi
+
+	sleep 0.1
+
+	set +e
+	echo kubectl auth can-i "${OPERATION}" "${RESOURCE}" "${TEST_NS}"
+	TEST=$(kubectl auth can-i "${OPERATION}" "${RESOURCE}" "${TEST_NS}")
+	echo "expected: ${EXPECTED}, actual: ${TEST}"
+	set -e
+	if [[ "${TEST}" == "${EXPECTED}" ]]; then
+		echo "----> PASSED"
+		return 0
+	fi
+
+	echo "----> |FAIL| Expected: ${EXPECTED}, Actual: ${TEST}"
+	return 1
+}
+
+function __testPermissionsClusterScoped() {
+	local OPERATION="$1"
+	local RESOURCE="$2"
+	local EXPECTED="$3"
+  __testPermissions "${OPERATION}" "${RESOURCE}" --all-namespaces "${EXPECTED}"
+}
+
+function __testDescribe() {
+	local RESOURCE="$1"
+	local TEST_NS="$2"
+	local EXPECTED="$3"
+
+	if [[ "${TEST_NS}" != "--all-namespaces" ]]; then
+		TEST_NS="-n${TEST_NS}"
+	fi
+
+	sleep 0.1
+
+	local result=1
+	set +e
+	kubectl describe "${RESOURCE}" "${TEST_NS}" > /dev/null
+	result=$?
+	set -e
+
+	local IS_OK="no"
```

Comment:
Why not `IS_OK=true`?

Reply:
I would rather suggest `IS_OK=false`, as in case of some different `EXPECTED` value that is not handled by IFs below, the script should not pass

Reply:
I would rather avoid using `false` and `true`. These are tricky and misleading in bash. Sometimes bash treats it is a builtin, sometimes as a command (/usr/bin/false), sometimes as a string. Take a look below. Surprise?
```
condition=false
if [ $condition ]; then
  echo 'Surprise!'
fi
```